### PR TITLE
ESP32: Add ESP32H2 target for lighting app

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
Using esp-idf v5.0-dev(commit-id 10f3aba770c316a4b56357435396b9cae094893e), we can build matter apps for ESP32H2.
Add ESP32H2 target for lighting app.

#### Change overview
* Platform changes and CHIPCryptoPALmbedTLS changes for idf v5.0-dev(10f3aba770)
  mbedtls in idf v5.0-dev(10f3aba770) has been updated to 3.0.1 so there is an issue #16561 
* Add ESP32H2 target for lighting app

#### Testing
Test on ESP32H2 with lighting app
```
$ cd $IDF_PATH
$ git checkout 10f3aba770
$ git submodule update --init --recursive
$ ./install.sh
$ . ./export.sh
$ cd connectedhomeip/examples/lighting-app/esp32
$ idf.py --preview set-target esp32h2
$ idf.py build
$ idf.py -p {ESPPORT} erase-flash flash monitor
```
Then we can provision it with chip tool `pairing ble-thread` command.

### CODE REVIEW

File: examples/lighting-app/esp32/main/OpenthreadLauncher.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/*
```
[the remark this answers is not in the document]
Comment (wqx6) [reply examples/lighting-app/esp32/main/OpenthreadLauncher.c]: Done

File: examples/lighting-app/esp32/main/CMakeLists.txt [hunk 2]
```diff
@@ -15,6 +15,11 @@
 #    limitations under the License.
 #
 # (Uses default behaviour of compiling all source files in directory, adding 'include' to include path.)
+if (NOT CONFIG_IDF_TARGET_ESP32H2)
```
[the remark this answers is not in the document]
Comment (wqx6) [reply examples/lighting-app/esp32/main/CMakeLists.txt]: Changed

File: config/esp32/components/esp32_mbedtls/mbedtls/esp_config.h [hunk 3]
```diff
@@ -32,9 +32,13 @@
 #ifndef ESP_CONFIG_H
 #define ESP_CONFIG_H
 
-#include "mbedtls/config.h"
 #include "sdkconfig.h"
-
+// For ESP32H2, we use idf v5.0, its "mbedtls/config.h" was replaced by "mbedtls/build_info.h"
+#if CONFIG_IDF_TARGET_ESP32H2
```
[the remark this answers is not in the document]
Comment (wqx6) [reply config/esp32/components/esp32_mbedtls/mbedtls/esp_config.h]: Changed

**Review by woody-apple — APPROVED** [review]
Fast tracking platform changes.

File: examples/lighting-app/esp32/README.md [hunk 4]
```diff
@@ -103,10 +114,20 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
 ## Commissioning over BLE using chip-tool
 
 -   Please build the standalone chip-tool as described [here](../../chip-tool)
--   Commissioning the Lighting device
+-   Commissioning the WiFi Lighting devices(ESP32, ESP32C3, ESP32S3)
 
         $ ./out/debug/chip-tool pairing ble-wifi 12345 <ssid> <passphrase> 20202021 3840
 
+-   For ESP32H2, please firstly start OpenThread Border Router on
```
Comment (chshu) [inline examples/lighting-app/esp32/README.md:121]:
```suggestion
-   For ESP32-H2, please firstly start OpenThread Border Router, you can either use [Raspberry Pi Openthread Border Router](https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/openthread_border_router_pi.md) OR [ESP Openthread Border Router](https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_br).
```

**Review by chshu — APPROVED** [review]

**Review by chshu — APPROVED** [review]

**Pull request merged**